browser_history/generic.py

```diff
@@ -268,3 +272,33 @@ def to_csv(self):
             for row in self.get():
                 writer.writerow(row)
             return output.getvalue()
+
+    def to_json(self, json_lines=True):
+        """
+         Return history formatted as a JSON Lines or Plain JSON format
+         names
+         :return: :py:class:`str` object
+        """
+        # custom json encoder for datetime objects
+        class DateTimeEncoder(json.JSONEncoder):
+            # Override the default method
+            def default(self, obj):
+                if isinstance(obj, (datetime.date, datetime.datetime)):
+                    return obj.isoformat()
+
+        # fetch lines
+        lines = []
+        for entry in self.entries:
+            json_record = {}
+            for field, value in zip(self.fields, entry):
+                json_record[field] = value
+            lines.append(json_record)
+
+        # if json_lines flag is true convert to JSON Lines format,
+        # otherwise convert it to Plain JSON format
+        if json_lines:
+            json_string = '\n'.join([json.dumps(line, cls=DateTimeEncoder) for line in lines])
+        else:
+            json_string = json.dumps({'history': lines}, cls=DateTimeEncoder)
```

Comment:
You can use the `indent` parameter of [json.dumps](https://docs.python.org/3/library/json.html#json.dumps) to get a much cleaner JSON (currently all the output is on a single line).

Reply:
(This is only for the regular json, not json lines)

Reply:
sure